### DISCUSSION TITLE
'client_secret' is plaintext

### DISCUSSION THREAD

**Discussion by stanthewizzard** [authelia/authelia#6885] [Q&A] [answered]
Hello

Since v4.38

There is this message:
Configuration: identity_providers: oidc: clients: client 'portainernetwork10': option 'client_secret' is plaintext but for clients not using the 'token_endpoint_auth_method' of 'client_secret_jwt' it should be a hashed value as plaintext values are deprecated with the exception of 'client_secret_jwt' and will be removed in the near future"

1°
I don't understand how it will work for services.
Exemple > WikiJS
the password is in plain text on the auth plugin

2°
how can we generate client_secret_jwt
hashed password ?

Thanks for explanation

**Comment by glime-ai** [comment]
Please note this response was created using generative AI leveraging previously available information from the repository. This information may be helpful however it is our strong recommendation that you carefully consider the information before blindly applying it.

---

Hello,

1. For services like WikiJS where the client secret is stored in plaintext in the authentication plugin configuration, it's important to note that Authelia is moving away from supporting plaintext secrets due to security concerns. The warning you're seeing is part of this transition. Services will need to adapt by either supporting hashed secrets or by using the `client_secret_jwt` method, which allows for a different approach to client authentication that doesn't require storing the secret in plaintext.

2. To generate a `client_secret_jwt`, you would typically use a tool or library that supports JWT creation. The `client_secret_jwt` is not simply a hashed password but a JWT that is signed with the client's secret. This JWT is then used as the client secret when making requests to the authorization server. Authelia has provided a way to generate client secrets and their hashes through its CLI tool. For example, you can use the `authelia crypto hash generate pbkdf2` command to generate a hash of your client secret. However, for `client_secret_jwt`, you would generate a JWT signed with your client secret and use that JWT as the `client_secret` in your configuration.

It's important to follow the best practices and guidelines provided by Authelia for generating and storing client secrets securely. If your service or plugin does not currently support these methods, you may need to reach out to the developers or look for updates that add support for more secure handling of client secrets.

For more detailed instructions on generating client secrets and configuring them securely, please refer to the Authelia documentation and the specific guides for generating client identifiers or secrets.

This comment was generated by [Glime](https://glimelab.ai/?utm_source=6fsHkcyufjQLaB&utm_medium=github).

**Comment by james-d-elliott** [comment, marked as answer]
The client must use this method for authenticating, it is solely responsible for generating and including it in the access request like any other credential. For Authelia as it is the provider and not the client, the passwords must be hashed using the available commands. 

The recommendations for performing this are highlighted in the FAQ that's linked on every OpenID Connect 1.0 integration guide pretty much:

https://www.authelia.com/integration/openid-connect/frequently-asked-questions/#how-do-i-generate-a-client-identifier-or-client-secret

**Comment by stanthewizzard** [comment]
thank you very very much :)

Maybe you will update integration for Komga or others to go away from plain pass ?

**Reply by james-d-elliott** [reply]
It's already updated, it was months ago.

**Reply by stanthewizzard** [reply]
so the integration that is here 
https://www.authelia.com/integration/openid-connect/komga/

Is accurate ? but it's plaintext ?
It will no work in the coming months ?
(sorry to bother you. Hard to understand)

**Comment by stanthewizzard** [comment]
I manage to do it with komga
YEAAAAH

Thanks again (I hope to be futur proof)
